### PR TITLE
MB - ItemViewModel.kt implementation

### PR DESCRIPTION
**Pull request by Mass-14**
ItemViewModel.kt implemented on the example of InventoryViewModel.kt

### CODE REVIEW

**Review by LozanoCyrielle — CHANGES_REQUESTED** [review]
You return the first item of the list not the one we are searching for
Maybe use the item id to find the one we are searching for
Also you there is no function to create/ to add an item to the database

**Comment by Valentin889** [comment]
The get item function needs an item id for it to make sense
You need a function to add a new item (this need an update in the db class too)
and a function to edit an existing item which also ned an update in the db class

**Comment by Valentin889** [comment]
That's the idea for update Item you got it (:
It's important to format before you push because first it will pass the CI and also it's way easier to see you change you actually made, review the diff like this is way harder

**Comment by Mass-14** [comment]
Thank you @Valentin889, I thought the format was needed only when adding new files...

**Comment by Valentin889** [comment]
1. The createItem function in db should take an item as a parameter; otherwise, the information you're going to input into the database won't make sense.

2. Instead of that, I would rather take the ID of an item as a parameter or the entire item itself, and filter based on that, not with the UI.
```
  private fun getItem() {
    viewModelScope.launch {
      database.getItems {
        for (i in it) {
          if (i.id == _uiState.value.item.id) {
            update(i)
          }
        }
      }
    }
  }
  ```
3. Be careful with the names, like update and updateItem, it's not clear what the difference is.
4. Some functions in the view model are public so that the UI can use them. If everything is private, the view model doesn't make much sense. Try to think about the use cases of each function (which class will use them and for what purpose). It will help to understand which parameter to take and properly name the functions.

File: app/src/main/java/com/android/partagix/model/ItemViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+              "", // no itemId exists at this moment, it will be generated
+              // and overwritten by the database
+              _uiState.value.item.category,
+              _uiState.value.item.name,
+              _uiState.value.item.description)
+
+      database.createItem("Yp5cetHh3nLGMsjYY4q9" /* todo get the correct userId */, newItem)
+    }
+  }
+
```
Comment (LozanoCyrielle) [inline app/src/main/java/com/android/partagix/model/ItemViewModel.kt:69]: Maybe more edit item instead of save item

**Review by LozanoCyrielle — APPROVED** [review]
I would just maybe rename the saveItem into Edit item but except that, everything seems good :))

**Review by Valentin889 — APPROVED** [review]
Seems good to me

**Pull request merged**